tests/test_syslog_settings.rb

```diff
@@ -41,14 +43,22 @@ def no_syslogsettings
 
   def test_syslogsettings_create
     syslog_setting = Cisco::SyslogSettings.new('default')
-    assert_includes(Cisco::SyslogSettings.syslogsettings, 'default')
-    assert_equal(Cisco::SyslogSettings.syslogsettings['default'],
-                 syslog_setting)
-
-    syslog_setting.timestamp = 'milliseconds'
-    assert_equal(Cisco::SyslogSettings.syslogsettings['default'].timestamp,
-                 'milliseconds')
-    assert_equal(syslog_setting.timestamp,
-                 'milliseconds')
+
+    if platform == :ios_xr
+      assert_nil(syslog_setting.timestamp)
+      assert_raises(Cisco::UnsupportedError) do
+        syslog_setting.timestamp = 'milliseconds'
+      end
+    else
+      assert_includes(Cisco::SyslogSettings.syslogsettings, 'default')
+      assert_equal(Cisco::SyslogSettings.syslogsettings['default'],
+                   syslog_setting)
+
+      syslog_setting.timestamp = 'milliseconds'
+      assert_equal(Cisco::SyslogSettings.syslogsettings['default'].timestamp,
+                   'milliseconds')
+      assert_equal(syslog_setting.timestamp,
+                   'milliseconds')
```

Comment:
I know this was in the original code but... the assert_equal args are swapped. Should be `assert_equal(expected, actual)`. Please fix that. :+1:  everything else.